libraries/AP_DDS/AP_DDS_External_Odom.cpp

```diff
@@ -0,0 +1,72 @@
+
+
+#include "AP_DDS_External_Odom.h"
+#include "AP_DDS_Type_Conversions.h"
+
+#if AP_DDS_VISUALODOM_ENABLED
+
+#include <AP_VisualOdom/AP_VisualOdom.h>
+#include <GCS_MAVLink/GCS.h>
+
+void AP_DDS_External_Odom::handle_external_odom(const tf2_msgs_msg_TFMessage& msg)
+{
+    auto *visual_odom = AP::visualodom();
+    if (visual_odom == nullptr) {
+        return;
+    }
+
+    for (size_t i = 0; i < msg.transforms_size; i++) {
+        const auto ros_transform_stamped = msg.transforms[i];
```

Comment:
can this be a reference?

Reply:
Done 977aba179814ad1633b3ca128135441fbb711fb3